src/sc_kpm/utils/search_utils.py

```diff
@@ -16,37 +16,34 @@
 
 
 def get_set_elements(set_node: ScAddr) -> List[ScAddr]:
-    elements = []
-    for target in (sc_types.LINK_VAR, sc_types.NODE_VAR):
-        templ = ScTemplate()
-        templ.triple(set_node, sc_types.EDGE_ACCESS_VAR_POS_PERM, target)
-        search_results = client.template_search(templ)
-        for result in search_results:
-            elements.append(result.get(2))
+    templ = ScTemplate()
+    templ.triple(set_node, sc_types.EDGE_ACCESS_VAR_POS_PERM, sc_types.UNKNOWN)
+    search_results = client.template_search(templ)
+    elements = [result.get(2) for result in search_results]
+    elements.reverse()  # Elements are found in reverse order
     return elements
 
 
 def get_oriented_set_elements(set_node: ScAddr) -> List[ScAddr]:
-    def get_next_element(access_edge: ScAddr = None):
-        if access_edge:
-            elem_search_result = _search_next_element_template(set_node, access_edge)
-        else:
-            keynodes = ScKeynodes()
-            elem_search_result = search_role_relation_template(set_node, keynodes[CommonIdentifiers.RREL_ONE.value])
-        if elem_search_result is None:
-            return None
-        elements.append(elem_search_result.get(ScAlias.ELEMENT.value))
-        return elem_search_result.get(ScAlias.ACCESS_EDGE.value)
-
     elements = []
-    elements_count = get_set_power(set_node)
```

Comment:
Are you sure that we can remove calling of this method? Write tests for empty sets to ensure about it

Reply:
Yeah. We needn't get power of set if we seach only one rrel edge and if it don't exist function returns